x-pack/plugins/security_solution/public/entity_analytics/components/entity_store/components/engines_status/index.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, Fragment } from 'react';
+import {
+  EuiLoadingSpinner,
+  EuiPanel,
+  EuiSpacer,
+  EuiButtonEmpty,
+  EuiFlexItem,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiCallOut,
+} from '@elastic/eui';
+import { capitalize } from 'lodash/fp';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { useErrorToast } from '../../../../../common/hooks/use_error_toast';
+import { downloadBlob } from '../../../../../common/utils/download_blob';
+import { useEntityEngineStatusNew } from '../../hooks/use_entity_engine_status';
+import { EngineComponentsStatusTable } from './components/engine_components_status';
+
+const FILE_NAME = 'engines_status.json';
+
+export const EngineStatus: React.FC = () => {
+  const { data, isLoading, error } = useEntityEngineStatusNew({ withComponents: true });
+
+  const downloadJson = useCallback(() => {
```

Comment:
do we need a useCallback for this? this could just be a pure utility function I think

Reply:
It is a good practice to always use `useCallback` on properties that are passed down to third-party components. Because they could use the property and a dependency for another hook like `useEffect` or be wrapped on `Reac.memo`.

But in this scenario it is unnecessary. I checked the Eui code. I will update this line!